Client/Frontend/Browser/Playlist/Cache/PlaylistCacheLoader.swift

```diff
@@ -0,0 +1,466 @@
+        "video/mpg": "mpg",
+        "video/mpeg": "mpeg",
+        "video/x-mpg": "mpg",
+        "video/x-mpeg": "mpeg",
+        "video/avi": "avi",
+        "video/x-m4v": "m4v",
+        "video/mp2t": "ts",
+        "application/vnd.apple.mpegurl": "mpg", // actually .m3u8, .m3u HLS stream
+        "video/3gpp": "3gp",
+        "text/vtt": "vtt",  // Subtitles format
+        "application/mp4": "mp4",
+        "application/x-mpegurl": "mpg", // actually .m3u8, .m3u HLS stream
+        "video/webm": "webm",
+        "application/ogg": "ogg",
+        "video/msvideo": "avi",
+        "video/x-msvideo": "avi",
+        "video/x-ms-wmv": "wmv",
+        "video/x-ms-wma": "wma",
+        "application/x-shockwave-flash": "swf",
+        "video/x-flv": "flv",
+        "video/x-mng": "mng",
+        "video/x-ms-asx": "asx",
+        "video/x-ms-asf": "asf",
+        "video/matroska": "mkv"
+    ]
+}
+
+class PlaylistWebLoader: UIView, WKNavigationDelegate {
+    private let tab = Tab(configuration: WKWebViewConfiguration().then {
+        $0.processPool = WKProcessPool()
+        $0.preferences = WKPreferences()
+        $0.preferences.javaScriptCanOpenWindowsAutomatically = false
+        $0.allowsInlineMediaPlayback = true
+        $0.ignoresViewportScaleLimits = true
+        $0.mediaTypesRequiringUserActionForPlayback = []
+    }, type: .private).then {
+        $0.createWebview()
+        $0.webView?.scrollView.layer.masksToBounds = true
+    }
+    
+    private var handler: (PlaylistInfo?) -> Void = { _ in }
+    private var handlerDidExecute = false
+    
+    init(handler: @escaping (PlaylistInfo?) -> Void) {
+        super.init(frame: .zero)
+
+        guard let webView = tab.webView else {
+            handlerDidExecute = true
+            handler(nil)
+            return
+        }
+        
+        self.handler = { [weak self] in
+            self?.handlerDidExecute = true
+            handler($0)
+        }
+        
+        self.addSubview(webView)
+        webView.snp.makeConstraints {
+            $0.edges.equalToSuperview()
+        }
+        
+        webView.navigationDelegate = self
+        tab.contentBlocker.setupTabTrackingProtection()
+        tab.addContentScript(tab.contentBlocker, name: ContentBlockerHelper.name(), sandboxed: false)
+        tab.addContentScript(FingerprintingProtection(tab: tab), name: FingerprintingProtection.name(), sandboxed: false)
+        tab.addContentScript(WindowRenderHelperScript(tab: tab), name: WindowRenderHelperScript.name(), sandboxed: false)
+        tab.addContentScript(PlaylistHelper(tab: tab), name: PlaylistHelper.name(), sandboxed: true)
+        tab.addContentScript(PlaylistWebLoaderContentHelper(self), name: PlaylistWebLoaderContentHelper.name(), sandboxed: false)
+        
+        let script: WKUserScript? = {
+            guard let path = Bundle.main.path(forResource: "PlaylistDetector", ofType: "js"), let source = try? String(contentsOfFile: path) else {
```

Comment:
I think we can sandbox this script as well, using `WKUserScript.createInDefaultContentWorld` - What do you think?